File: 0191_Number_of_1_Bits/solution.java
package cn.huang.leetcode;

/*
191. Number of 1 Bits
 */
public class LC_0191_Numberof1Bits {
    public int hammingWeight(int n) {
        int count = 0;
        for (int i = 0; i < 32; i++) {
            if (((n >> i) & 1) == 1) {
                count++;
            }
        }
        return count;
    }
}

File: 0191_Number_of_1_Bits/solution.cpp
/*
191. Number of 1 Bits

Level: Easy

https://leetcode.com/problems/number-of-1-bits
*/
#include <vector>
#include <string>
#include <unordered_map>
#include <unordered_set>

#include <algorithm>

#include "common/ListNode.h"
#include "common/TreeNode.h"

using namespace std;

/*
  Solution: 
*/

class Solution
{
public:
  int hammingWeight(uint32_t n)
  {
    int count;

    while (n > 0)
    {
      if (n & 1)
      {
        count++;
      }
      n = n >> 1;
    }
    return count;
  }

  int hammingWeight_v2(uint32_t n)
  {
    int count;

    for (int i =0 ; i< 32 ; i++)
    {
      if (n & (1 << i))
      {
        count++;
      }

    }
    return count;
  }
};